packages/backend/src/services/SavedSqlService/SavedSqlService.ts

```diff
@@ -0,0 +1,226 @@
+import {
+    CreateSqlChart,
+    ForbiddenError,
+    SessionUser,
+    SqlChart,
+    UpdateSqlChart,
+} from '@lightdash/common';
+import { LightdashAnalytics } from '../../analytics/LightdashAnalytics';
+import { ProjectModel } from '../../models/ProjectModel/ProjectModel';
+import { SavedSqlModel } from '../../models/SavedSqlModel';
+import { SpaceModel } from '../../models/SpaceModel';
+import { BaseService } from '../BaseService';
+
+type SavedSqlServiceArguments = {
+    analytics: LightdashAnalytics;
+    projectModel: ProjectModel;
+    spaceModel: SpaceModel;
+    savedSqlModel: SavedSqlModel;
+};
+
+export class SavedSqlService extends BaseService {
+    private readonly analytics: LightdashAnalytics;
+
+    private readonly projectModel: ProjectModel;
+
+    private readonly spaceModel: SpaceModel;
+
+    private readonly savedSqlModel: SavedSqlModel;
+
+    constructor(args: SavedSqlServiceArguments) {
+        super();
+        this.analytics = args.analytics;
+        this.projectModel = args.projectModel;
+        this.spaceModel = args.spaceModel;
+        this.savedSqlModel = args.savedSqlModel;
+    }
+
+    async getSqlChart(
+        user: SessionUser,
+        projectUuid: string,
+        savedSqlUuid: string,
+    ): Promise<SqlChart> {
+        const savedChart = await this.savedSqlModel.get(savedSqlUuid, {
+            projectUuid,
+        });
+        const space = await this.spaceModel.getSpaceSummary(
+            savedChart.space.uuid,
+        );
+        const access = await this.spaceModel.getUserSpaceAccess(
+            user.userUuid,
+            savedChart.space.uuid,
+        );
+
+        if (
+            user.ability.cannot(
+                'view',
+                subject('SavedChart', {
+                    organizationUuid: savedChart.organization.organizationUuid,
+                    projectUuid: savedChart.project.projectUuid,
+                    isPrivate: space.isPrivate,
+                    access,
+                }),
+            )
+        ) {
+            throw new ForbiddenError("You don't have access to this chart");
+        }
+        return savedChart;
+    }
+
+    async createSqlChart(
+        user: SessionUser,
+        projectUuid: string,
+        sqlChart: CreateSqlChart,
+    ): Promise<string> {
+        const { organizationUuid } = await this.projectModel.getSummary(
+            projectUuid,
+        );
+        if (
+            user.ability.cannot(
+                'manage',
+                subject('CustomSql', { organizationUuid, projectUuid }),
+            )
+        ) {
+            throw new ForbiddenError();
+        }
+        const space = await this.spaceModel.getSpaceSummary(sqlChart.spaceUuid);
+        const { isPrivate } = space;
+        const access = await this.spaceModel.getUserSpaceAccess(
+            user.userUuid,
+            sqlChart.spaceUuid,
+        );
+        if (
+            user.ability.cannot(
+                'create',
```

Comment:
The FE lets them send these requests for now. This access check means its 'safe' but we can disable saving them if they done have this.